### PR TITLE
Update snapshot tests with dev rlang and dev testthat

### PR DESCRIPTION
**Pull request by DavisVaughan**
- First commit moves to dev rlang
- Second commit then adds dev testthat on top of that

### CODE REVIEW

File: tests/testthat/_snaps/chop.md [hunk 1]
```diff
@@ -4,29 +4,29 @@
...
+      Error in `df_unchop()`: `ptype` must be a named list.
 
 # incompatible sizes are caught
 
     Code
       (expect_error(unchop(df, c(x, y))))
     Output
       <error/rlang_error>
-      In row 1, can't recycle input of size 2 to size 3.
+      Error in `fn()`: In row 1, can't recycle input of size 2 to size 3.
```
Comment (DavisVaughan) [inline tests/testthat/_snaps/chop.md:23]: This is unfortunate, it comes from using `purrr::reduce()` so I think it is out of our control

File: tests/testthat/_snaps/pivot-wide.md [hunk 2]
```diff
@@ -98,5 +98,5 @@
       (expect_error(pivot_wider(df, values_fn = 1)))
     Output
       <error/rlang_error>
-      Can't convert a double vector to function
+      Error in `map()`: Can't convert `.x[[i]]`, a double vector, to a function.
```
Comment (DavisVaughan) [inline tests/testthat/_snaps/pivot-wide.md:101]: Interesting change here

File: tests/testthat/_snaps/rectangle.md [hunk 3]
```diff
@@ -4,79 +4,79 @@
...
+      Error in `hoist()`: `.data` must be a data frame.
 
 # unnest_wider - bad inputs generate errors
 
     Code
       (expect_error(unnest_wider(df, y)))
     Output
       <error/rlang_error>
-      Column `y` must contain a list of vectors.
+      Error in `.f()`: Column `y` must contain a list of vectors.
```
Comment (DavisVaughan) [inline tests/testthat/_snaps/rectangle.md:57]: Comes from doing a `map(out, elt_to_wide)` where `elt_to_wide()` looks like `.f()`

**Review by DavisVaughan — COMMENTED** [review]
Ultimately I don't think this looks too bad. There are a few `stop_vctrs()` calls being shown, but I don't think it is so many that I'd want to spend a bunch of time right now updating them to be more informative if a future vctrs release is going to fix that.

Many of the calls already reference an internal tidyr function, or the top level tidyr verb, which is pretty good

File: tests/testthat/_snaps/chop.md (hunk 1 above)
Comment (lionel-) [reply tests/testthat/_snaps/chop.md:23]: We could probably pass a lambda to avoid this.

File: tests/testthat/_snaps/pivot.md [hunk 4]
```diff
@@ -15,21 +15,21 @@
       (expect_error(check_spec(df)))
     Output
       <error/rlang_error>
-      The `.name` column must be a character vector.
+      Error in `check_spec()`: The `.name` column must be a character vector.
```
Comment (lionel-) [inline tests/testthat/_snaps/pivot.md:18]: This should set the error call.

File: tests/testthat/_snaps/chop.md [hunk 5]
```diff
@@ -4,29 +4,29 @@
       (expect_error(chop(df)))
     Output
       <error/rlang_error>
-      Argument `cols` is missing with no default
+      Error in `check_present()`: Argument `cols` is missing with no default
```
Comment (lionel-) [inline tests/testthat/_snaps/chop.md:7]: This should set the error call, or maybe use the rlang function.

File: tests/testthat/_snaps/pivot-wide.md [hunk 6]
```diff
@@ -52,23 +52,23 @@
...
+      Error in `build_wider_spec()`: `values_from` must select at least one column.
 
 # `values_fn` emits an informative error when it doesn't result in unique values (#1238)
 
     Code
       (expect_error(pivot_wider(df, values_fn = list(value = ~.x))))
     Output
       <error/rlang_error>
-      Applying `values_fn` to `value` must result in a single summary value per key.
+      Error in `vals_dedup()`: Applying `values_fn` to `value` must result in a single summary value per key.
```
Comment (lionel-) [inline tests/testthat/_snaps/pivot-wide.md:63]: This should set the error call.

File: tests/testthat/_snaps/rectangle.md [hunk 7]
```diff
@@ -4,79 +4,79 @@
...
     Output
       <error/rlang_error>
-      `.col` must identify a list-column.
+      Error in `hoist()`: `.col` must identify a list-column.
     Code
       (expect_error(df %>% hoist(x, 1)))
     Output
       <error/rlang_error>
-      All elements of `...` must be named.
+      Error in `check_pluckers()`: All elements of `...` must be named.
```
Comment (lionel-) [inline tests/testthat/_snaps/rectangle.md:28]: This should set the error call.

File: tests/testthat/_snaps/rectangle.md [hunk 8]
```diff
@@ -107,23 +107,23 @@
...
+      Error in `unnest_wider()`: `data` must be a data frame.
 
 # unnest_longer - bad inputs generate errors
 
     Code
       (expect_error(unnest_longer(df, y)))
     Output
       <error/rlang_error>
-      Column `y` must contain a list of vectors.
+      Error in `elt_to_long()`: Column `y` must contain a list of vectors.
```
Comment (lionel-) [inline tests/testthat/_snaps/rectangle.md:126]: This should set the error call.

File: tests/testthat/_snaps/replace_na.md [hunk 9]
```diff
@@ -4,15 +4,15 @@
       (expect_error(replace_na(1, 1:10)))
     Output
       <error/rlang_error>
-      Replacement for `data` is length 10, not length 1.
+      Error in `check_replacement()`: Replacement for `data` is length 10, not length 1.
```
Comment (lionel-) [inline tests/testthat/_snaps/replace_na.md:7]: This should set the error call.

**Review by lionel- — COMMENTED** [review]
I agree it doesn't look too bad overall. We can already improve messages in many cases by setting the error call in internal helpers and passing lambdas to functionals.

**Pull request closed without merging**